### PR TITLE
ft: add cart feature(view/add/delete)

### PR DESCRIPTION
**Pull request by shyakadavis**
# Description

🍏 Delivers #184361231-#184361229-#184361230 where: a User/Buyer 👤  should be able to:
- add items to their cart
- view their cart
- clear their cart

Extended checks are against the situations when user:
- inserts an invalid UUID (isValidUUID() utility check)
- tries to add a quantity that exceeds the stock capacity
- tries to add a product that does't exist
- wants to remove an item from the cart
- wants to view their cart total

Fixes # (issue)

# Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have created a loom video for the new feature or enhancement 
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by freddykaberuka — APPROVED** [review]

File: src/utils/calculateCartTotal.util.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+export const calculateCartTotal = async (itemsInCart: CartAttributes[]) => {
+  try {
+    const items = await JSON.parse(JSON.stringify(itemsInCart));
+    // iterate the items in cart and get the product details (we want the price)
+    let cartTotal = 0;
+
+    for (const item of items) {
+      const itemTotal = item.Product.price * item.quantity;
+      cartTotal += itemTotal;
+    }
```
Comment (mucyoalexaxel) [inline src/utils/calculateCartTotal.util.ts:17]:
```suggestion
```
Refactor this logic using `.reduce()` method on arrays. It is more readable.

File: src/utils/isValidUUID.util.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// a function to check if a string is a valid UUID
+// example of a valid UUID: 3fa85f64-5717-4562-b3fc-2c963f66afa6
+// example of where this function is used: src/controllers/Cart.controller.ts 📐 line 20
```
Comment (mucyoalexaxel) [inline src/utils/isValidUUID.util.ts:3]:
```suggestion
```

File: test/cart.spec.ts [hunk 3]
```diff
@@ -0,0 +1,195 @@
+import request from 'supertest';
```
[the remark this answers is not in the document]
Comment (mucyoalexaxel) [reply test/cart.spec.ts:1]: fix this error @shyakadavis

**Review by mucyoalexaxel — CHANGES_REQUESTED** [review]
@shyakadavis 
Great work on this PR. have a look at some of the suggested changes and keep me posted.

File: .hound.yml [hunk 4]
```diff
@@ -1,2 +1,4 @@
 eslint:
   enabled: true
+tslint:
+  enabled: true
```
Comment (mucyoalexaxel) [inline .hound.yml:4]:
```suggestion
eslint:
  enabled: true
  config_file: .eslintrc.json
```
Try this.

File: src/utils/isValidUUID.util.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export const isValidUUID = (uuid: string) => {
+  const uuid_regex =
```
Comment (mucyoalexaxel) [inline src/utils/isValidUUID.util.ts:2]:
```suggestion
  const uuidRegex =
```

File: tsconfig.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     "target": "es2016",
-    "module": "commonjs",
+    "module": "ESNext",
```
Comment (mucyoalexaxel) [inline tsconfig.json:4]:
```suggestion
    "module": "commonjs",
```

**Review by mucyoalexaxel — CHANGES_REQUESTED** [review]
@shyakadavis 
Try the suggested changes.

File: .hound.yml (hunk 4 above)
Comment (shyakadavis) [reply .hound.yml:4]: Let me take a look.

**Review by mucyoalexaxel — APPROVED** [review]

**Pull request merged**